daemon/start.go

```diff
@@ -11,6 +11,15 @@ func (daemon *Daemon) ContainerStart(job *engine.Job) error {
 	if len(job.Args) < 1 {
 		return fmt.Errorf("Usage: %s container_id", job.Name)
 	}
+
+	config := runconfig.ContainerConfigFromJob(job)
+	hostConfig := runconfig.ContainerHostConfigFromJob(job)
```

Comment:
How come you changed this logic around?
Also, you should be able to pass a nil config as such: `validateVolumePath(nil, hostConfig)` and it'll be just fine.

We should really keep the old logic since a `hostConfig` may or may not be sent on `start`, if there is one passed in, then it needs to be set with `setHostConfig`.